src/data/ability.ts

```diff
@@ -2259,13 +2259,17 @@ export class CheckTrappedAbAttr extends AbAttr {
     super(false);
   }
   
-  applyCheckTrapped(pokemon: Pokemon, passive: boolean, trapped: Utils.BooleanHolder, args: any[]): boolean | Promise<boolean> {
+  applyCheckTrapped(pokemon: Pokemon, passive: boolean, trapped: Utils.BooleanHolder, otherPokemon: Pokemon[], args: any[]): boolean | Promise<boolean> {
     return false;
   }
 }
 
 export class ArenaTrapAbAttr extends CheckTrappedAbAttr {
-  applyCheckTrapped(pokemon: Pokemon, passive: boolean, trapped: Utils.BooleanHolder, args: any[]): boolean {
+  applyCheckTrapped(pokemon: Pokemon, passive: boolean, trapped: Utils.BooleanHolder, otherPokemon: Pokemon[], args: any[]): boolean {
+    if (otherPokemon[0].getTypes().indexOf(Type.GHOST) != -1){
```

Comment:
Could be simplified as
```suggestion
    if (otherPokemon[0].getTypes().includes(Type.GHOST)){
```

Reply:
Also, how does this work with more than one Pokemon on the field? From a cursory glance it looks like you're only checking the first party member?

Reply:
> Could be simplified as

Agreed - made change

Reply:
> Also, how does this work with more than one Pokemon on the field? From a cursory glance it looks like you're only checking the left party member?

`otherPokemon`, while an array, ever only contains a single Pokemon. It's the Pokemon that is being targeted by a trapping move or the Pokemon that is trying to run/switch. That's not something I wanted to do; that is how it is in `phases.ts` and it's just passed on to `ability.ts`.

Here's it working on right side Pokemon too
https://github.com/pagefaultgames/pokerogue/assets/85713900/540e985b-6981-4da0-8edc-9b8f813bbb69

Reply:
Figured out why it was passing in as an array and it is now updated to not do that 👍